### PR TITLE
SI-5717 error when bytecode cannot be written

### PR DESCRIPTION
**Pull request by som-snytt**
If there's an existing file foo when emitting a class file for foo.Bar,
then emit an error at foo.Bar, similar to what javac does.

The error message looks something like:

foo.scala:4: error: error writing object Foo: ./mypkg/Foo.class: ./mypkg is not a directory

This is a non-destructive version of https://github.com/scala/scala/pull/1746

It's worth quoting https://github.com/scala/scala/pull/2258

"We don't need to assert our way out of tight spots, we can issue an error."

Review by @paulp who went this way before.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala [hunk 1]
```diff
@@ -454,7 +455,7 @@ abstract class GenASM extends SubComponent with BytecodeWriters with GenJVMASM {
     }
 
     // -----------------------------------------------------------------------------------------
-    // utitilies useful when emitting plain, mirror, and beaninfo classes.
+    // utilities useful when emitting plain, mirror, and beaninfo classes.
     // -----------------------------------------------------------------------------------------
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala:459]: Got something against utitilies?

**Comment by paulp** [comment]
This LGTM except the commit message. See https://github.com/scala/scala/wiki/Pull-Request-Policy re the subject. It could be something like

```
SI-5717 error when bytecode cannot be written
```

File: src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala (hunk 1 above)
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala:459]: It was kind of titillating.

**Comment by som-snytt** [comment]
Thanks, I thought my commit messages were impeccable but I was wrong.  Mio peccato.

**Comment by paulp** [comment]
Thanks for the rescue.

**Comment by som-snytt** [comment]
I just got this error message, 16 months after the last time.  Man, is that satisfying.

I had a junk.scala containing scala-user junk in an eponymous package, because I couldn't believe Xlint had nothing better to say than, "I'm an option, not a mind reader!"  But I also had an old file junk.

**Pull request merged**